android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -196,36 +197,40 @@ protected WebView createViewInstance(ThemedReactContext reactContext) {
 
-        DownloadManager.Request request = new DownloadManager.Request(Uri.parse(url));
-
-        String fileName = URLUtil.guessFileName(url, contentDisposition, mimetype);
-        String downloadMessage = "Downloading " + fileName;
-
-        //Attempt to add cookie, if it exists
-        URL urlObj = null;
         try {
-          urlObj = new URL(url);
-          String baseUrl = urlObj.getProtocol() + "://" + urlObj.getHost();
-          String cookie = CookieManager.getInstance().getCookie(baseUrl);
-          request.addRequestHeader("Cookie", cookie);
-          System.out.println("Got cookie for DownloadManager: " + cookie);
-        } catch (MalformedURLException e) {
-          System.out.println("Error getting cookie for DownloadManager: " + e.toString());
-          e.printStackTrace();
-        }
+          DownloadManager.Request request = new DownloadManager.Request(Uri.parse(url));
+
+          String fileName = URLUtil.guessFileName(url, contentDisposition, mimetype);
+          String downloadMessage = "Downloading " + fileName;
+
+          //Attempt to add cookie, if it exists
+          URL urlObj = null;
+          try {
+            urlObj = new URL(url);
+            String baseUrl = urlObj.getProtocol() + "://" + urlObj.getHost();
+            String cookie = CookieManager.getInstance().getCookie(baseUrl);
+            request.addRequestHeader("Cookie", cookie);
+            System.out.println("Got cookie for DownloadManager: " + cookie);
+          } catch (MalformedURLException e) {
+            System.out.println("Error getting cookie for DownloadManager: " + e.toString());
+            e.printStackTrace();
+          }
 
-        //Finish setting up request
-        request.addRequestHeader("User-Agent", userAgent);
-        request.setTitle(fileName);
-        request.setDescription(downloadMessage);
-        request.allowScanningByMediaScanner();
-        request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
-        request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, fileName);
+          //Finish setting up request
+          request.addRequestHeader("User-Agent", userAgent);
+          request.setTitle(fileName);
+          request.setDescription(downloadMessage);
+          request.allowScanningByMediaScanner();
+          request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
+          request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, fileName);
 
-        module.setDownloadRequest(request);
+          module.setDownloadRequest(request);
 
-        if (module.grantFileDownloaderPermissions()) {
-          module.downloadFile();
+          if (module.grantFileDownloaderPermissions()) {
+            module.downloadFile();
+          }
+        } catch (IllegalArgumentException e) {
+          System.out.println("IllegalArgumentException: " + e.toString());
```

Comment:
This code fits in with the error-handling philosophy found in the rest of the method, which is to just ignore errors quietly and just log them. In that sense, this change seems fine.

Outside the scope of this PR: It seems to me like there should be some kind of callback added like `onDownloadFailedToStart`, which would alert the client of this library that a file download has failed to start.

Reply:
@trcoffman thanks for you comment. Good point is to add `onDownloadFailedToStart` callback. But I've fixed concrete case here which affected us with crashes in production. Callback implementation should be done separately I suppose.

Reply:
Are you going to release it soon?

Reply:
We have already publish it in our own fork https://www.npmjs.com/package/@pandadoc/react-native-webview.